src/me/mrCookieSlime/Slimefun/listeners/ArmorListener.java

```diff
@@ -32,48 +34,73 @@ public ArmorListener(SlimefunPlugin plugin) {
-										EntityDamageByEntityEvent event = new EntityDamageByEntityEvent(p, n, DamageCause.ENTITY_ATTACK, e.getDamage() / 2);
-										Bukkit.getPluginManager().callEvent(event);
-										if (!event.isCancelled()) ((LivingEntity) n).damage(e.getDamage() / 2);
-									}
-								}
-							}
-							for (int i = 0; i < 2; i++) {
-								for (BlockFace face: BlockFace.values()) {
-									p.getWorld().playEffect(p.getLocation().getBlock().getRelative(BlockFace.DOWN).getRelative(face).getLocation(), Effect.STEP_SOUND, p.getLocation().getBlock().getRelative(BlockFace.DOWN).getRelative(face).getType());
+					else if (item.isItem(SlimefunItems.BOOTS_OF_THE_STOMPER)
+						&& Slimefun.hasUnlocked(p,
+						SlimefunItems.BOOTS_OF_THE_STOMPER, true)
+						&& e.getCause() == DamageCause.FALL
+					) {
+						e.setCancelled(true);
+						p.getWorld().playSound(p.getLocation(), Sound.ENTITY_ZOMBIE_BREAK_WOODEN_DOOR, 2F, 2F);
+						p.setVelocity(new Vector(0.0, 0.7, 0.0));
+						for (Entity n: p.getNearbyEntities(4, 4, 4)) {
+							if (n instanceof LivingEntity && !n.getUniqueId().equals(p.getUniqueId())) {
+								n.setVelocity(n.getLocation().toVector().subtract(p.getLocation().toVector())
+									.normalize().multiply(1.4));
+								if (p.getWorld().getPVP()
+									&& SlimefunPlugin.getProtectionManager()
+									.hasPermission(p, n.getLocation(), ProtectionModule.Action.PVP)
+								) {
+									EntityDamageByEntityEvent event = new EntityDamageByEntityEvent(p, n,
+										DamageCause.ENTITY_ATTACK, e.getDamage() / 2);
+									Bukkit.getPluginManager().callEvent(event);
+									if (!event.isCancelled()) ((LivingEntity) n).damage(e.getDamage() / 2);
 								}
 							}
 						}
+
+						for (BlockFace face: BlockFace.values()) {
+							Block b = p.getLocation().getBlock().getRelative(BlockFace.DOWN).getRelative(face);
+							p.getWorld().playEffect(b.getLocation(), Effect.STEP_SOUND, b.getType());
+						}
 					}
-					else if (SlimefunItem.getByItem(armor).isItem(SlimefunItems.SLIME_BOOTS_STEEL) && Slimefun.hasUnlocked(p, SlimefunItems.SLIME_BOOTS_STEEL, true) && e.getCause() == DamageCause.FALL) e.setCancelled(true);
+					else if (item.isItem(SlimefunItems.SLIME_BOOTS_STEEL)
+						&& Slimefun.hasUnlocked(p, SlimefunItems.SLIME_BOOTS_STEEL, true)
+						&& e.getCause() == DamageCause.FALL
+					)
+						e.setCancelled(true);
 				}
 			}
 		}
 	}
 	
 	@EventHandler
 	public void onTrample(PlayerInteractEvent e) {
-		if (e.getAction() == Action.PHYSICAL && e.getClickedBlock().getType() == Material.FARMLAND && SlimefunManager.isItemSimiliar(e.getPlayer().getInventory().getBoots(), SlimefunItem.getItem("FARMER_SHOES"), true)) e.setCancelled(true);
+		if (e.getAction() == Action.PHYSICAL
+			&& e.getClickedBlock() != null
+			&& e.getClickedBlock().getType() == Material.FARMLAND
+			&& SlimefunManager.isItemSimiliar(e.getPlayer().getInventory().getBoots(),
+			SlimefunItem.getItem("FARMER_SHOES"), true)
```

Comment:
You should use SlimefunItems.FARMER_SHOES instead, SlimefunItem.getItem() is a heavy call that involves a loop.